main.go

```diff
@@ -74,25 +74,37 @@ func main() {
 		quit := make(chan struct{})
 		go func(host string, metric *prometheus.GaugeVec) {
 			resultPrev := ""
-
+			resultPrevErr := ""
 			for {
 				select {
 				case <-ticker.C:
 					result, err := net.LookupHost(host)
 
 					if err != nil {
-						metric.With(prometheus.Labels{"host": host, "status": "success", "result": resultPrev}).Set(0)
-						metric.With(prometheus.Labels{"host": host, "status": "failed", "result": err.Error()}).Set(1)
+						resultErr := err.Error()
+						fmt.Printf("Error: %s\n", resultErr)
+						
+						metric.With(prometheus.Labels{"host": host, "status": "failed", "result": resultErr}).Set(1)
+						if(resultPrev != ""){
```

Comment:
Is this if switch needed?

Reply:
This is the same as the one above, in case where the first is not successful there will be an metric with status success and an empty result with value 0 that will never change